qbft/timeout.go

```diff
@@ -1,18 +1,84 @@
 package qbft
 
 import (
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/bloxapp/ssv-spec/types"
 	"github.com/pkg/errors"
-	"time"
+	"golang.org/x/exp/constraints"
 )
 
 var (
-	quickTimeoutThreshold = Round(8)        //nolint
-	quickTimeout          = 2 * time.Second //nolint
-	slowTimeout           = 2 * time.Minute //nolint
+	quickTimeoutThreshold = Round(8) //nolint
+	// quickTimeout is the timeout in seconds for the first 8 rounds
+	quickTimeout int64 = 2 // 2 seconds
+	// slowTimeout is the timeout in seconds for rounds after the first 8
+	slowTimeout int64 = 120 // 2 minutes
 	// CutoffRound which round the instance should stop its timer and progress no further
 	CutoffRound = 15 // stop processing instances after 8*2+120*6 = 14.2 min (~ 2 epochs)
 )
 
+type RoundTimer struct {
+	// role is the role of the QBFT instance
+	Role types.BeaconRole
+	// height is the current height of the instance
+	Height Height
+	// network is the beacon network
+	Network types.BeaconNetwork
+	//current unix epoch time in seconds
+	CurrentTime int64
+}
+
+func (t *RoundTimer) TimeoutForRound(round Round) int64 {
+	switch t.Role {
+	case types.BNRoleAttester, types.BNRoleSyncCommittee:
+		return max(AttestationOrSyncCommitteeTimeout(round, t.Height, t.Network)-t.CurrentTime, 0)
+	case types.BNRoleAggregator, types.BNRoleSyncCommitteeContribution:
+		return max(AggregationOrContributionTimeout(round, t.Height, t.Network)-t.CurrentTime, 0)
```

Comment:
different than impl

not sure if this can happen even if the system clock is bad...